warp10/src/main/java/io/warp10/json/GTSEncoderSerializer.java

```diff
@@ -0,0 +1,86 @@
+//
+//   Copyright 2020  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.geoxp.GeoXPLib;
+import io.warp10.continuum.gts.GTSDecoder;
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.continuum.store.thrift.data.Metadata;
+
+import java.io.IOException;
+
+public class GTSEncoderSerializer extends JsonSerializer<GTSEncoder> {
+
+  @Override
+  public void serialize(GTSEncoder encoder, JsonGenerator gen, SerializerProvider provider) throws IOException {
+    Metadata metadata = encoder.getMetadata();
+    String name = metadata.getName();
+    if (null == name) {
+      name = "";
+    }
+
+    gen.writeStartObject();
+    gen.writeStringField("c", name);
+    gen.writeObjectField("l", metadata.getLabels());
+    gen.writeObjectField("a", metadata.getAttributes());
+    gen.writeNumberField("la", metadata.getLastActivity());
+    gen.writeFieldName("v");
+    gen.writeStartArray();
+
+    GTSDecoder decoder = encoder.getUnsafeDecoder(false);
+    while (decoder.next()) {
+      long ts = decoder.getTimestamp();
+      long location = decoder.getLocation();
+      long elevation = decoder.getElevation();
+      // We do not call getBinaryValue because JSON cannot represent byte arrays
```

Comment:
You're right.
If you're fine with the b64 serialization of byte[] we could call `getBinaryValue` here and b64 encode if the returned value it `byte[]`.
However, in the case of a GTS, we're stuck with an ISO-8859-1 string, correct me if I'm wrong.

Reply:
I am not too fond of emitting b64 content for byte arrays, why not simply output an ISO-8859-1 string?

Reply:
This would unify the GTS, Encoder and byte[] representation for binary data which is nice.
There is a slight downside as it will produce a lot of special characters which will be escaped in JSON, making that not very compact. I think it's fine because JSON shouldn't be used for large transfer of binary data.